### DISCUSSION TITLE
npm publish should fail when the `files` is misconfig in package.json

### DISCUSSION THREAD

**Discussion by aladdin-add** [npm/feedback#425] [CLI] [answered]
background: package.json
```json
{
  "files": [
    "conf",
    "lib",
    "template",
    "vendor"
  ],
}
```
![image](https://user-images.githubusercontent.com/13050025/120637265-44eb9280-c4a1-11eb-8c24-ccd361b9df64.png)

and lately I renamed the `template` => `templates`, but forgot to update the files. it can be published successfully.
I argue it should (at lease) output a warning, as there is no file matched the `"template"`.

**Comment by MylesBorins** [comment, marked as answer]
@aladdin-add thanks for opening this discussion. If you would like to make a concrete proposal to change the npm CLI the next step is to open an issue at https://github.com/npm/rfcs
